d2common/path_tile.go

```diff
@@ -13,6 +13,7 @@ type PathTile struct {
 	UpLeft, UpRight     *PathTile
 	DownLeft, DownRight *PathTile
 	Position            d2vector.Position
+	SubX, SubY          int
```

Comment:
`SubX` and `SubY` are equal to `Position.Floor().X()` and `Y()` (or maybe that vector+ 1,1 - see `Position.RenderOffset`).

`Position` used to contain methods for these things but I removed them as they were unused. Since the sub tile position is used here, I suppose we should add a `Position.SubTile` method to get that value.